### DISCUSSION TITLE
Import samples quickly and efficiently

### DISCUSSION THREAD

**Discussion by maxtribe** [bangcorrupt/hacktribe#73] [Q&A] [answered]
Hello people

I urgently need your help. At the weekend I installed the Hacktribe firmware and tried to import my samples with the e2sSample.all file. My problem, however, is that the samples are neither loaded at startup nor when I go to the Data Utility and load the Import All Samples file there. I am always shown Unknown Error. I really only care about three things. That the names of the samples are displayed and that it is quick so that I don't always have to do everything individually. I've already heard that only the samples were named 501 to 510 on the restart, then on the restart they were renamed to Sample510 and so on.

Can one of you help me? Would be very grateful for that

Kind regards Max

**Comment by maks** [comment]
Not sure if it's the same problem as your are seeing, but I had similar (same?) error message when I loaded samples at too low number of slots. I think I started mine from 521 upwards to get it to work.

**Reply by bangcorrupt** [reply]
Since d219d6b loading samples to slots >=501 should work without issues.  Let me know if that's not the case.

**Comment by bangcorrupt** [comment, marked as answer]
Hi, you need to edit the e2sSample.all file so the first sample is in slot 501.  There is a free editor, [Oe2sSLE](https://github.com/JonathanTaquet/Oe2sSLE).  

Before importing the e2sSample.all file, click '[Import Options](https://github.com/JonathanTaquet/Oe2sSLE#144-import-options)' in the bottom right of the [main window](https://raw.githubusercontent.com/JonathanTaquet/Oe2sSLE/master/doc/images/screenshot_main.png) and select '501' in the drop down menu labeled 'From #Num'.  

Then click '[Import e2sSample.all](https://github.com/JonathanTaquet/Oe2sSLE#142-import-samples-of-a-sample-library-file)' and load your file.  The samples should be loaded starting at slot 501.  Then just save the file and put it in the `Sample` directory of the SD card.

**Reply by maxtribe** [reply]
@bangcorrupt Should I turn off the Electribe when I wanna load the file

**Reply by bangcorrupt** [reply]
Once edited, you should be able to use 'Import All Samples' from the Data Utility. If the file is in the `Sample` directory, it will be loaded at boot.

Rebooting is probably a good idea if you got an error message when it booted this time.

**Reply by Maxklui** [reply]
@bangcorrupt I added my samples to the e2Sample.all file and the all file was also displayed in the data utility. I pressed import all samples and nothing happened and I also selected slot 501 when importing. I'm doing something wrong. I use the Oe2sSLE tool

**Reply by bangcorrupt** [reply]
Does it work if you import [this e2sSample.all](https://github.com/bangcorrupt/hacktribe-content/raw/master/carved_beats_set/e2sSample.all)? 

Can you share an e2sSample.all that doesn't work?

**Reply by maxtribe** [reply]
@bangcorrupt I went there, formatted the SD card in the Electribe and then went to Export All Samples to create the e2sSamples.all file

**Reply by BKLronin** [reply]
@maxtribe Try to downlaod the e2s form above, it contains a "factory" bank so you can test if you re correctly importing it.

The samples are within the 2s file, you dont have to copy the samples seperately or anything. They then get loaded on start.

**Reply by maxtribe** [reply]
@BKLronin Do you mean the e2sSample.all file??

**Reply by maxtribe** [reply]
@bangcorrupt @BKLronin It's worked out. Finally. But one more thing. How can I then import samples afterwards, because I have now loaded samples and when I have them again, I want to import them again in the same principle

**Reply by BKLronin** [reply]
Have a look at the sampler manual from Korg for basic operation. :)

**Reply by Maxklui** [reply]
@BKLronin If I want to import new samples, then theoretically I would have to delete the existing e2sSample.all file, right? And that's exactly what I want to avoid, because I have now occupied slots 501 to 524 and then want to occupy slot 525 upwards over time

**Reply by maxtribe** [reply]
@BKLronin Is there something that makes this possible afterwards to add more samples to the already imported samples or do I have to buy several SD cards

Thank you so much

**Reply by BKLronin** [reply]
You can load your es2sample.all file anytime in the editor on pc and add new samples at the end.

**Reply by Maxklui** [reply]
@BKLronin Can I then do the whole thing with another tool such as the Electribe 2 Sample Manager from Appification? Because the python tool would be too cumbersome for me in the long run. And this Tool is so much easier to use for me

**Reply by BKLronin** [reply]
Try it out :) Should work, can´t remember if it caused problems or not. The python one lets you import whole directories if you set it up correctly. The sample manager you had to import file by file.

**Reply by Maxklui** [reply]
@BKLronin In the Electribe 2 Sample Manager you can double-click on the WAV files that you want to import, then it works really fast.

**Reply by BKLronin** [reply]
Complete folders?

**Reply by Maxklui** [reply]
@BKLronin @bangcorrupt I have an last general question: Will there actually be updates to the Hacktribe firmware or not?

**Reply by bangcorrupt** [reply]
Hacktribe will continue to be updated. Anything labeled `enhancement` is something I have some idea how to approach.  Some of these may turn into features some day.  I'm going to work on making existing features more accessible before trying to add any more new ones.  It will be cool when we can trade FX presets.

**Reply by maks** [reply]
@Maxklui thats a great idea! I would love to have that too and I think I actually saw someone else ask for this previously, though not sure how hard (or even possible) this will be to implement, I'm sure @bangcorrupt will know much more about that.

**Reply by Maxklui** [reply]
Yes i hope so. @maks do you maybe know how can I install the enhacement on the Electribe

**Reply by maks** [reply]
Likewise, I'm really excited by bangcorrupts awesome work here, its really opened up so many new ways for me to use my e2s - its such a shame Korg hasn't open sourced the firmware to make these efforts much easier.

**Reply by Maxklui** [reply]
@maks I know what do you mean. Do you maybe know how can I install Sysex on the Electribe?

**Reply by maks** [reply]
@Maxklui Im sorry Im not sure what you mean by: install Sysex on the Electribe. This is maybe getting a bit of topic for this discussion thread. Did you want to start a new discussion for that question and "at" me and we can continue there.

**Reply by Maxklui** [reply]
@bangcorrupt I wanted to ask what updates and features are coming for the hacktribe? And what functions has the Updates included

**Reply by bangcorrupt** [reply]
See [previous comment](https://github.com/bangcorrupt/hacktribe/discussions/73#discussioncomment-1964302).  [Discussions](https://github.com/bangcorrupt/hacktribe/discussions?discussions_q=label%3Aenhancement) and [issues](https://github.com/bangcorrupt/hacktribe/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement) can be filtered by label.  There's no guarantee any of these will ever work.

**Comment by coderofsalvation** [comment]
Perhaps in the README.md or 'How to'-wiki article, this:

*Apply patch to Electribe Sampler firmware version 2.02 only.*

could be changed into:

*Apply patch to Electribe Sampler firmware version 2.02 only.*
> note: existing sampler livesets (`e2sSample.all`) need to be refactored (using [Oe2sSLE](https://github.com/JonathanTaquet/Oe2sSLE) e.g.) since samples below sampleslot 501 will get ignored (to allow the new features).

^------ This will warn future sampler-users about the pros/cons.

(I tried forking the repo to do it myself, but it seems the howto-wiki article is the most important place afaik)

**Reply by bangcorrupt** [reply]
I updated the wiki.  By the way, editing the e2sSample.all file won't refactor the sample numbers in the pattern files.  See https://github.com/bangcorrupt/hacktribe/discussions/61#discussioncomment-4502488 for a script that edits allpat files.
